modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/CoreComponentBuildingDefinitionProvider.java

```diff
@@ -0,0 +1,422 @@
+    public static final String DEFAULT_EXCEPTION_STRATEGY = "default-exception-strategy";
+    public static final String NAME_EXCEPTION_STRATEGY_ATTRIBUTE = "globalName";
+    public static final String CUSTOM_EXCEPTION_STRATEGY = "custom-exception-strategy";
+    public static final String CHOICE_EXCEPTION_STRATEGY = "choice-exception-strategy";
+    public static final String SET_PAYLOAD = "set-payload";
+    public static final String PROCESSOR_CHAIN = "processor-chain";
+    public static final String PROCESSOR = "processor";
+    public static final String TRANSFORMER = "transformer";
+    public static final String FILTER = "filter";
+    public static final String CUSTOM_PROCESSOR = "custom-processor";
+    public static final String CLASS_ATTRIBUTE = "class";
+    public static final String SUB_FLOW = "sub-flow";
+    public static final String RESPONSE = "response";
+    public static final String MESSAGE_FILTER = "message-filter";
+    public static final String FLOW = "flow";
+    public static final String EXCEPTION_LISTENER_ATTRIBUTE = "exceptionListener";
+    public static final String SCATTER_GATHER = "scatter-gather";
+    public static final String WIRE_TAP = "wire-tap";
+    public static final String ENRICHER = "enricher";
+    public static final String ASYNC = "async";
+    public static final String TRANSACTIONAL = "transactional";
+    public static final String UNTIL_SUCCESSFUL = "until-successful";
+    public static final String FOREACH = "foreach";
+    public static final String FIRST_SUCCESSFUL = "first-successful";
+    public static final String ROUND_ROBIN = "round-robin";
+    public static final String CHOICE = "choice";
+    public static final String OTHERWISE = "otherwise";
+    public static final String ALL = "all";
+    public static final String POLL = "poll";
+
+    private ComponentBuildingDefinition.Builder baseDefinition;
+    private ComponentBuildingDefinition.Builder transactionManagerBaseDefinition;
+
+    @Override
+    public void init(MuleContext muleContext)
+    {
+        baseDefinition = new ComponentBuildingDefinition.Builder().withNamespace(CORE_NAMESPACE_NAME);
+        transactionManagerBaseDefinition = baseDefinition.copy();
+    }
+
+    @Override
+    public List<ComponentBuildingDefinition> getComponentBuildingDefinitions()
+    {
+
+        LinkedList<ComponentBuildingDefinition> componentBuildingDefinitions = new LinkedList<>();
+
+
+        AttributeDefinition messageProcessorListAttributeDefinition = fromChildListConfiguration(MessageProcessor.class).build();
+        ComponentBuildingDefinition.Builder exceptionStrategyBaseBuilder = baseDefinition.copy()
+                .withSetterParameterDefinition(MESSAGE_PROCESSORS, messageProcessorListAttributeDefinition)
+                .withSetterParameterDefinition("globalName", fromSimpleParameter(NAME).build());
+        componentBuildingDefinitions.add(baseDefinition.copy()
```

Comment:
it seems like it's always needed to invoke copy() on the definition. That is very error prone and untidy. Looks like this DSL needs another pass